### PR TITLE
Added addProduct page and some other stuff

### PR DESCRIPTION
**Pull request by okapifan**

### CODE REVIEW

File: app/Http/Controllers/ProductController.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    public function create()
+    {
+        return view('addproduct');
+    }
+
+    public function store(){
+        $product = new Product;
+        $product->name = request('productname');
```
Comment (TijmenWierenga) [inline app/Http/Controllers/ProductController.php]: To increase readability of variables I advise you to _camel case_ your variables like `productName` and `productBrand`

**Review by TijmenWierenga — APPROVED** [review]

File: app/Product.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace App;
+
+use Illuminate\Database\Eloquent\Model;
+
+class Product extends Model
+{
+    public function user() {
```
Comment (mbicknese) [inline app/Product.php]: Not psr-2 compliant.

File: app/Http/Controllers/ProductController.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    public function create()
+    {
+        return view('addproduct');
+    }
+
+    public function store(){
+        $product = new Product;
+        $product->name = request('productName');
```
Comment (mbicknese) [inline app/Http/Controllers/ProductController.php]: Please validate and sanitize all incoming data.

File: app/User.php [hunk 4]
```diff
@@ -27,4 +27,12 @@ class User extends Authenticatable
         'password', 'remember_token', 'active',
     ];
 
+    public function product() {
```
Comment (mbicknese) [inline app/User.php]: Psr-2

File: routes/web.php [hunk 5]
```diff
@@ -13,6 +13,13 @@
 
 Auth::routes();
 
-Route::get('/home', 'HomeController@index')->name('home');
+Route::get('/home', function () {
+    return redirect()->route('products.index');
+})->name('home');
+
+Route::get('/products/index', 'ProductController@index')->name('products.index');
```
Comment (mbicknese) [inline routes/web.php]: Drop the `/index` from the URL

**Review by mbicknese — CHANGES_REQUESTED** [review]

File: app/Product.php (hunk 2 above)
Comment (okapifan) [reply app/Product.php]: What do you mean with psr-2??

Comment (mbicknese) [reply app/Product.php]: http://www.php-fig.org/psr/psr-2/

File: app/Http/Controllers/ProductController.php [hunk 6]
```diff
@@ -36,13 +37,23 @@ public function create()
...
-    public function store(){
+    public function store(Request $request){
+
         $product = new Product;
+
+        $request->validate([
+            'productName' => 'required|unique:products|string',
+            'productBrand' => 'nullable|string',
+            'productPictureUrl' => 'nullable|active_url',
+            'added_by' => 'integer',
```
Comment (mbicknese) [inline app/Http/Controllers/ProductController.php:48]: Not used as you get the id from the authenticated user, right? (which is a good thing)

Comment (okapifan) [reply app/Http/Controllers/ProductController.php:48]: When you add a product he fills the added_by in with the id of the user that is logged in at that moment. Then he goes with the rest of the data to the validator and than he adds it to the database

Comment (okapifan) [reply app/Http/Controllers/ProductController.php:48]: I don't know if that answers your answer somewhere

**Review by mbicknese — APPROVED** [review]

**Pull request merged**